pkg/client/k8s/v1alpha1/unstructured.go

```diff
@@ -264,8 +265,56 @@ func UpdateNamespace(n string) UnstructuredMiddleware {
 	}
 }
 
+// AddKubeServerVersionToLabels adds kubernetes server version to instance's
+// labels
+//
+// TODO
+// Move this override flag to a Predicate. This will follow idiomatic Maya
+// convention to separate conditional(s) logic from core business logic. This
+// in turn helps in readability & maintainability of the codebase.
+//
+// e.g.
+// func IsLabelSet(key string) UnstructuredPredicate {...}
+// func IsLabelUnSet(key string) UnstructuredPredicate {...}
+func AddKubeServerVersionToLabels(override bool) UnstructuredMiddleware {
+	return func(given *unstructured.Unstructured) (updated *unstructured.Unstructured) {
+		if given == nil {
+			return given
+		}
+		l := given.GetLabels()
+		if l == nil {
+			l = map[string]string{}
+		}
+		// TODO
+		// error should not be ignored here
+		// Need to design a custom struct that wraps unstructured.Unstructured
+		// instance and has fields to accomodate error(s), warning(s), etc
+		//
+		// NOTE:
+		//  This will be done along with idiomatic maya refactorings
+		vInfo, err := GetServerVersion()
```

Comment:
Fetching of the `K8s serverVersion` can be moved inside the `if` condition at `line#299` as the labels will only be changed when the `override` is set to `true` or it isn't set.